packages/block-editor/src/hooks/utils.js

```diff
@@ -115,6 +118,21 @@ export function shouldSkipSerialization( blockType, featureSet, feature ) {
 	return skipSerialization;
 }
 
+export function useStyleOverride( style ) {
+	const { setStyleOverride, deleteStyleOverride } = unlock(
+		useDispatch( blockEditorStore )
+	);
+	const fallbackId = useId();
+	useEffect( () => {
+		if ( ! style ) return;
+		const id = style.id || fallbackId;
+		setStyleOverride( id, style );
+		return () => {
+			deleteStyleOverride( id );
+		};
+	}, [ fallbackId, style, setStyleOverride, deleteStyleOverride ] );
```

Comment:
`style` is an object and `useStyleOverride` is typically called with a fresh object in each call, so it looks like this `useEffect` is firing too frequently. E.g. if I log out just before `setStyleOverride` while scrolling the editor, it looks like it gets called a lot:

https://github.com/WordPress/gutenberg/assets/14988353/7ea0ff35-5edb-4664-a3c5-b8735aa93367

Would it work to destructure the `style` object before the `useEffect` and then use each of those values in the `useEffect` dependency array, before calling `setStyleOverride` with a new object? Perhaps something like the following (I haven't tried this, so just an idea):

```js
	const { css, id, assets, __unstableType } = style;

	useEffect( () => {
		if ( ! css && ! assets ) return;
		const _id = id || fallbackId;
		setStyleOverride( _id, { assets, css, id, __unstableType } );
		return () => {
			deleteStyleOverride( id );
		};
	}, [ assets, css, fallbackId, id, setStyleOverride, deleteStyleOverride, __unstableType ] );
```

I might be missing some details there, though!